### PR TITLE
refactor: five seam-tightening consolidations (day-1 review PR 7/7)

### PR DESCRIPTION
**Pull request by mnbf9rca**
## Summary

Five seam-tightening consolidations identified in the day-1 review. Each task is self-contained; the changes share only the "PR 7" label.

- **Tests reference `BackupFile.schemaURL`** instead of inlining the literal across 5 call sites — schema URL is now single-source-of-truth (finding 9).
- **opaque-swift UniFFI generator writes directly into `Sources/`**, eliminating the `generated/ → Sources/` copy step and its silent-overwrite drift risk (findings 18 + 37). Build script reorganized so `Sources/` reaches final state in one contiguous block before xcframework packaging.
- **`AuthenticationCoordinatorView` no longer reaches through `viewModel.lockStateService`**; the VM now exposes `onEnterBackground()` and `shouldLockOnForeground`, and `lockStateService` is `private` (finding 34).
- **`LockStateService` init no longer back-writes `defaultTimeout`** to UserDefaults; the getter's existing `≤0` fallback is now the only mechanism, so future changes to `defaultTimeout` reach upgrading users (finding 35).
- **`BackupFileService` password policy aligned to `PasswordValidationService.minimumLength` (12 chars, NIST SP 800-63B)** — security tightening from the previous 8-char floor. Backups leave the device, so applying a weaker policy than the in-app validator was counter-intuitive (finding 38).

References: `docs/day-1-review/2026-04-18-detailed-report.md` findings 9, 18, 34, 35, 37, 38.

## Design notes for reviewers

- **`PasswordValidationService.minimumLength` is referenced via the concrete type, not via a protocol property.** This is intentional — `BackupFileService` does not hold a `PasswordValidationServiceProtocol` instance, and the plan explicitly chose the direct-reference approach (Option A) over injection (Option B). A `PasswordPolicy` shared type would be YAGNI at this stage. If `BackupFileService` later needs compound validation, lifting `minimumLength` onto the protocol is the natural follow-up.
- **opaque-swift build script reorder** (commit fe99a83) moves the `mv`/`rm` cleanup and `module.modulemap` heredoc to immediately follow bindgen, before lipo + xcframework. Functionally equivalent; readability win.
- **`AuthenticationViewModelScenePhaseTests` partially overlaps with `AuthenticationCoordinatorViewTests`** by design — VM unit vs coordinator integration. Doc comment on the new file (commit 2cf16a4) makes this explicit.

## Test plan

- [x] New tests added per task: `BackupFile.schemaURL` references compile (5 sites); `AuthenticationViewModelScenePhaseTests` (3 tests) cover the new VM surface; `LockStateServiceTests` gain `initDoesNotSeedStoredTimeout` (TDD red→green) and `defaultTimeoutChangeReachesUpgradingUsers`; `BackupFileServiceTests` gain parameterized `rejectsPasswordsBelowAlignedMinimum` (8 + 11 char) and `acceptsPasswordAtAlignedMinimum` (12 char).
- [x] `scripts/run-tests.sh` passes in CI (full iOS suite). **Local note**: full-suite runs reproduce ~24 pre-existing ViewInspector flakes also present on `main` (verified by bisect — failure list byte-identical between `main` and this branch). Tracked in #186; not introduced by this PR.
- [x] `scripts/check-coverage.sh` ≥85%.
- [x] `pre-commit run --all-files` clean.
- [x] `opaque-swift/build-xcframework.sh` rebuilds the xcframework cleanly; iOS clean build resolves the package; OpaqueAuthService tests pass.

## Follow-ups (out of scope for this PR)

- `LockStateService.lock()`, `logout()`, and the new `shouldLockOnForeground()` method all omit tracing per `coding-practices.md` for SwiftUI re-evaluation contexts, but none have a comment explaining why tracing is absent. Conformance is a file-wide cleanup, not piecemeal.
- `@MainActor` annotation consistency across VM scene-phase methods (existing `lock()` lacks the annotation; same applies to the new `onEnterBackground`). Address holistically rather than only on new methods.
- `BackupFileServiceTestHelpers` private enum is still inside the main test struct body; if `type_body_length` triggers again, it's the next candidate for extension extraction.

## Summary by Sourcery

Align backup password policy with shared minimum length, tighten lock state timeout behavior, encapsulate lock state access behind the authentication view model, and streamline UniFFI Swift binding generation into final SwiftPM sources while updating tests and CI accordingly.

Bug Fixes:
- Ensure backup encryption rejects passwords shorter than the app-wide minimum password length and accepts passwords at the aligned threshold.
- Prevent lock timeout initialization from writing default values into UserDefaults so updated defaults apply to existing users.

Enhancements:
- Centralize backup schema URL usage via BackupFile.schemaURL across backup tests to avoid hard-coded literals.
- Expose scene phase handling methods on AuthenticationViewModel and make its LockStateService dependency private, updating the coordinator view to use the new surface.
- Generate UniFFI Swift bindings directly into Sources/OpaqueSwiftFFI and Sources/OpaqueSwift, simplifying the xcframework build flow and avoiding an intermediate generated directory.
- Make PasswordValidationService.minimumLength publicly accessible as a static constant for reuse in password policy checks.

CI:
- Adjust iOS CI cache and artifact paths to remove the obsolete opaque-swift/generated directory.

Tests:
- Add backup password policy tests covering rejection below the aligned minimum and acceptance at the minimum length.
- Add lock state service tests verifying init does not seed timeout in UserDefaults and that default timeout changes reach upgrading users.
- Add AuthenticationViewModel scene phase delegation tests to verify interaction with LockStateService.
- Refactor BackupFileServiceTests using an extension to keep new password policy tests separate from format validation tests.

### CODE REVIEW

File: ios/FamilyMedicalApp/FamilyMedicalApp/ViewModels/Auth/AuthenticationViewModel.swift [hunk 1]
```diff
@@ -196,6 +196,16 @@ final class AuthenticationViewModel {
         await unlockWithBiometric()
     }
 
+    // MARK: - Scene Phase Actions
+
+    func onEnterBackground() {
+        lockStateService.recordBackgroundTime()
+    }
+
+    var shouldLockOnForeground: Bool {
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply ios/FamilyMedicalApp/FamilyMedicalApp/ViewModels/Auth/AuthenticationViewModel.swift]: Fixed in 8ca1ea5. Converted `var shouldLockOnForeground: Bool` → `func shouldLockOnForeground() -> Bool`. The underlying `LockStateService.shouldLockOnForeground()` calls `userDefaults.removeObject(forKey: backgroundTimeKey)` (mutates state); a method signals the side-effecting check explicitly at every call site. Single caller in `AuthenticationCoordinatorView` and 2 scene-phase tests updated.

File: ios/FamilyMedicalApp/FamilyMedicalAppTests/Services/Backup/BackupFileServiceTests.swift [hunk 2]
```diff
@@ -71,6 +71,36 @@ struct BackupFileServiceTests {
...
+    func acceptsPasswordAtAlignedMinimum() throws {
+        let service = BackupFileService(
+            keyDerivationService: KeyDerivationService(),
+            encryptionService: EncryptionService()
+        )
+        let payload = BackupFileServiceTestHelpers.makeTestPayload()
+
+        // Exactly 12 characters — must NOT throw.
+        _ = try service.createEncryptedBackup(payload: payload, password: "TwelveChar1!")
+    }
```
[the remark this answers is not in the document]
Comment (mnbf9rca) [reply ios/FamilyMedicalApp/FamilyMedicalAppTests/Services/Backup/BackupFileServiceTests.swift:108]: Valid concern, partially accepted with a twist. Fixed in 82b395a + polish in 83c6b03.

Pure dynamic fixtures alone would solve staleness on *upward* policy changes but create the opposite hazard: if `minimumLength` were *lowered* (security regression from 12 → 8), the dynamic rejection/acceptance fixtures would happily re-derive and the suite would pass — silently letting the floor erode.

Split the concern across two files:

- `BackupFileServiceTests`: fixtures now derive from `PasswordValidationService.minimumLength` via `String(repeating:count:)` — verifies enforcement of whatever the current policy is (no staleness on upward bumps).
- `PasswordValidationServiceTests`: new `minimumLengthEnforcesNISTFloor` test asserts `minimumLength >= 12` (NIST SP 800-63B §5.1.1.2 mandates ≥8; project policy sets 12 as the internal floor) — catches downward regression.

Concerns properly separated: backup tests verify backup behavior; `PasswordValidationServiceTests` owns the policy floor.
